spec/features/budgets/groups_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+feature 'Groups' do
+
+  let(:user)   { create(:user, :level_two) }
+  let(:budget) { create(:budget) }
+  let(:group)  { create(:budget_group, budget: budget) }
+
+  context 'show' do
+
+    scenario "group not found raises 404 error" do
+      expect { visit budget_group_path('s', 'n') }.to raise_error(ActionController::RoutingError)
+    end
+
+    scenario "Budget correctly found using slug" do
```

Comment:
This is good, but I would use `id` instead of `slug` in the scenario names, as that is what we are really testing 
To test the slug we would just pass as a parameter the budget object, instead of the id